doc/reference/s3gateway_api.md

```diff
@@ -49,39 +47,40 @@ hash of the file contents.
 
 #### GET Bucket location
 
-Route: `GET /<branch>.<repo>/?location`
+Route: `GET /<repo>/?location`
 
-This will always serve the same location for every bucket, but the endpoint is implemented to provide better compatibility with S3 clients.
+This will always serve the same location for every bucket, but the endpoint is
+implemented to provide better compatibility with S3 clients.
 
 #### List Multipart Uploads
 
-Route: `GET /<branch>.<repo>/?uploads`
+Route: `GET /<branch>.<repo>/?uploads&branch=<branch>`
 
-Lists the in-progress multipart uploads in the given branch. The `delimiter` query parameter is not supported.
+Lists the in-progress multipart uploads in the given branch. The `delimiter`
+query parameter is not supported. The `branch` parameter is a non-standard
+extension that allows you to specify which branch to on the given repo to
+list objects from; if unspecified, it defaults to `master`.
 
 #### PUT Bucket
 
-Route: `PUT /<branch>.<repo>/`.
+Route: `PUT /<repo>/`.
 
-If the repo does not exist, it is created. If the branch does not exist, it
-is likewise created. As per S3's behavior in some regions (but not all),
-trying to create the same bucket twice will return a `BucketAlreadyOwnedByYou`
-error.
+If the repo does not exist, it is created. As per S3's behavior in some
+regions (but not all), trying to create the same bucket twice will return a
```

Comment:
It's crazy that it only returns that error in some regions. What does it return in the other ones?

Reply:
us-east-1 has a few behavioral differences, including this. It was the first region, so I guess they fixed issues like this in subsequently released regions, but kept us-east-1 different for backwards compatibility.